pyramid/config/__init__.py

```diff
@@ -903,6 +921,13 @@ def end(self):
         """
         return self.manager.pop()
 
+    def __enter__(self):
+        self.begin()
+        return self
+
+    def __exit__(self, exc_type, exc_value, exc_traceback):
+        self.end()
```

Comment:
Well played sir. Well played.

> Just that any work set up to be done within the scope of the with block is cancelled.

This is reasonable to me. I do wish we had better examples in the docs for creating custom directives, but we can consider that separately. Want me to make this change using your example?